adapter/config/crd/conversion.go

```diff
@@ -61,7 +62,13 @@ func convertConfig(schema model.ProtoSchema, config model.Config) (IstioObject,
 	return out, nil
 }
 
-// camelCaseToKabobCase converts "my-name" to "MyName"
+// resourceName converts "my-name" to "myname".
+// This is needed by k8s API server as dashes prevent kubectl from accessing CRDs
+func resourceName(s string) string {
```

Comment:
Seems to be a very risky change - what is the set of valid chars for kubectl ? Can we reject names with '-' instead, 
and document they are not supported ? 

Many subtle bugs (including security, etc) happen when characters are removed from names.

Reply:
Bug in k8s that won't be fixed by the release time.